pkg/manager/member/failure_recovery.go

```diff
@@ -0,0 +1,324 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package member
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	"github.com/pingcap/tidb-operator/pkg/util"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// annAutoFailureRecovery in TidbCluster indicates whether auto failure recovery is enabled for the components
+	annAutoFailureRecovery = "kubernetes.io/auto-failure-recovery"
+
+	podPhaseRunning = "Running"
+	podPhaseUnknown = "Unknown"
+
+	// Reason for host down being true
+	hdReasonNodeFailure           = "NodeFailure"
+	hdReasonRODiskFound           = "RODiskFound"
+	hdReasonStoreDownTimeExceeded = "StoreDownTimeExceeded"
+
+	restartToDeleteStoreGap = 10 * time.Minute
+)
+
+// FailureObjectAccess contains the common set of functions to access the properties of a failure object
+type FailureObjectAccess interface {
```

Comment:
nit
```suggestion
// FailedObjectAccess contains the common methods to access the properties of a failed pod
type FailedObjectAccess interface {
```

Reply:
failure object here denotes a FailureStore or FailureMember and not a pod (Here Store and Member is generalized as "object")

Reply:
I see.

Reply:
I have explained this bit now in the comment as well.